From 557b60798a9da49386f1034b133838332735de22 Mon Sep 17 00:00:00 2001
From: Emilio Pozuelo Monfort <pochu@debian.org>
Date: Tue, 27 Dec 2016 17:24:10 +0100
Subject: [PATCH] Fix possible free of uninitialized pointer

If the _XReply() call fails, we'll try to free an uninitialized
pointer.

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=849026

Reported-by: Thomas Walker <thwalker3@gmail.com>
Signed-off-by: Emilio Pozuelo Monfort <pochu@debian.org>
Reviewed-by: Julien Cristau <jcristau@debian.org>
Tested-by: Thomas Walker <thwalker3@gmail.com>
Signed-off-by: Julien Cristau <jcristau@debian.org>
---
 src/XIQueryDevice.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/XIQueryDevice.c b/src/XIQueryDevice.c
index a877d78..51e2d0d 100644
--- a/src/XIQueryDevice.c
+++ b/src/XIQueryDevice.c
@@ -46,7 +46,7 @@ XIQueryDevice(Display *dpy, int deviceid, int *ndevices_return)
     char                *ptr;
     char                *end;
     int                 i;
-    char                *buf;
+    char                *buf = NULL;
 
     XExtDisplayInfo *extinfo = XInput_find_display(dpy);
 
-- 
2.8.3

